File: src/main/scala/org/mcphoton/command/StopCommand.java
package org.mcphoton.command;

import org.mcphoton.command.Command;
import org.mcphoton.messaging.Messageable;

/**
 * @author TheElectronWill
 */
public class StopCommand implements Command {
	@Override
	public void execute(Messageable source, String[] args) {
		System.exit(0);
	}

	@Override
	public String getDescription() {
		return "Stops the server";
	}

	@Override
	public String getName() {
		return "stop";
	}
}

File: src/main/scala/org/mcphoton/command/StopCommand.scala
package org.mcphoton.command

import org.mcphoton.messaging.Messageable
import org.mcphoton.runtime.ExecutionGroup
import org.mcphoton.server.PhotonServer
import org.mcphoton.world.World

/**
 * @author TheElectronWill
 */
class StopCommand extends Command("stop") {
	override def execute(sender: Messageable, args: Seq[String])
						(implicit w: World, exgroup: ExecutionGroup): Unit = {
		PhotonServer.shutdown()
	}
}
